modules/lang-painless/src/main/java/org/elasticsearch/painless/lookup/PainlessLookup.java

```diff
@@ -40,14 +40,16 @@
     private final Map<String, PainlessMethod> painlessMethodKeysToImportedPainlessMethods;
     private final Map<String, PainlessClassBinding> painlessMethodKeysToPainlessClassBindings;
     private final Map<String, PainlessInstanceBinding> painlessMethodKeysToPainlessInstanceBindings;
+    private final Map<String, Class<?>> methodNameToInjectedScriptType;
```

Comment:
Shouldn't this information come from the context (parsed into ScriptClassInfo)? This doesn't seem like the right place to track this information.

Reply:
I had it come from here because I wanted to use the type from the whitelist. I figured it could be an interface or a superclass of the actual script. I think something like this makes a lot more sense with your idea of injecting members though.